course_catalog/etl/loaders.py

```diff
@@ -0,0 +1,45 @@
+"""Course catalog data loaders"""
+from django.contrib.contenttypes.models import ContentType
+
+from course_catalog.models import Course, Program, ProgramItem
+
+
+def load_course(course_data):
+    """Load the course into the database"""
+    course_id = course_data.pop("course_id")
+    course, _ = Course.objects.update_or_create(
+        course_id=course_id, defaults=course_data
+    )
+    return course
+
+
+def load_program(program_data):
+    """Load the program into the database"""
+    program_id = program_data.pop("program_id")
+    courses_data = program_data.pop("courses")
+
+    program, _ = Program.objects.update_or_create(
+        program_id=program_id, defaults=program_data
+    )
+
+    for position, course_data in enumerate(courses_data):
+        # skip courses that don't define a course_id
+        if not course_data.get("course_id", None):
+            continue
+
+        course = load_course(course_data)
+
+        # create a program item or update its position
+        ProgramItem.objects.update_or_create(
+            program=program,
+            content_type=ContentType.objects.get(model="course"),
+            object_id=course.id,
+            defaults={"position": position},
+        )
```

Comment:
Again not sure how likely this is to happen if ever in production, but if I remove a course from a program in MM, then rerun the import task, the course is not removed from the `Program` model here.

Reply:
It's not impossible, I could definitely imagine over time perhaps a course is retired, or maybe even renumbered, in which case we'd want to keep the course for historical reasons, but remove the linkage here.